### ISSUE TITLE
Tests fail on Windows

### ISSUE THREAD

**Issue by derektom14** [willowtreeapps/assertk#296]
I recently cloned this project onto my Windows machine and ran the `jvmTest`s. The one change I made was to remove the `@OptIn(ExperimentalCoroutinesApi::class)` annotations because I'm on an earlier version of Kotlin, but I doubt that's responsible for these errors:

- `FileTest` has many tests that fail because they doesn't recognize the Windows path separator
- `JVMResultTest.failure_result_fails_with_stacktrace` fails because `errorLines` is split by `"\n"`, which isn't the right line separator on Windows
- `JavaAnyTest.isDataClassEqualTo_reports_all_properties_that_differ_on_failure` fails due to an extra CRLF.
- `PathTest.bytes_value_correct_byte_array_passes` fails, but it isn't clear why, as the values are cut off:
```
org.opentest4j.AssertionFailedError: 
Expected :...0, 0x00, 0x00, 0x00][]
Actual   :...0, 0x00, 0x00, 0x00][]
```

**Comment by derektom14** [comment]
When I start running `commonTest` I get many more examples, all appear to be due to path separators and newlines.

**Comment by evant** [comment]
Good catch, haven't really tested on windows.

**Comment by derektom14** [comment]
I can work on the PR to fix this, as I'm on Windows.

**Comment by derektom14** [comment]
PR to fix: #300
